homeassistant/components/fan/__init__.py

```diff
@@ -45,6 +45,7 @@
 SPEED_OFF = 'off'
 SPEED_LOW = 'low'
 SPEED_MEDIUM = 'medium'
+SPEED_MED = 'med'
```

Comment:
Right.  Short story:  I'm a colossal knee-biter.  I thought I was insulated from 'med' or 'medium' because my automations use templates.  So that the devices now discovered was only one half of the issue.  Somewhere in the back of my head I knew what's being sent is numbers, not the actual 'med' or 'medium', but sheesh.

Anyway.  There is no problem #5817 is everything and good enough.  I humbly apologize and withdraw the PR.


Long story:
So, I'm using automations like this:
```yaml
  - alias: Set fan speed
    trigger:
      platform: state
      entity_id: input_select.master_bedroom_fanmotor
    action:
      service: fan.set_speed
      data_template:
        entity_id: fan.master_bedroom_fanmotor
        speed: '{{ trigger.to_state.state | lower }}'
```
with another one that sets the input_select if the fan is changed from the wall and so figured I was good.  I don't even ever say 'med' or 'medium' (or low or high or whatever).  It just flows through.  Yeah.  That's because I'm an idiot.  It's over in the input_select.

I went and changed the input_select and everything thing is just peachy now.

I'm a knob.  That's why you guys are the members and I'm a contributor.  Sorry for wasting your time and thanks for putting up with me!